pkg/apis/apps/validation/validation_test.go

```diff
@@ -366,267 +364,169 @@ func TestValidateStatefulSet(t *testing.T) {
-				},
-			},
+			set: mkStatefulSet(&validPodTemplate,
+				tweakReplicas(3),
+				tweakUpdateStrategy(mkUpdateStrategy(
+					tweakMaxUnavailable(intstr.FromInt(0)),
+				)),
+			),
 			errs: field.ErrorList{
 				field.Invalid(field.NewPath("spec", "updateStrategy", "rollingUpdate", "maxUnavailable"), nil, ""),
 			},
 		},
 		{
 			name: "zero percent maxUnavailable",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.ParallelPodManagement,
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					UpdateStrategy: apps.StatefulSetUpdateStrategy{
-						Type: apps.RollingUpdateStatefulSetStrategyType,
-						RollingUpdate: &apps.RollingUpdateStatefulSetStrategy{
-							MaxUnavailable: intStrAddr(intstr.FromString("0%")),
-						},
-					},
-				},
-			},
+			set: mkStatefulSet(&validPodTemplate,
+				tweakReplicas(3),
+				tweakUpdateStrategy(mkUpdateStrategy(
+					tweakMaxUnavailable(intstr.FromString("0%")),
+				)),
+			),
 			errs: field.ErrorList{
 				field.Invalid(field.NewPath("spec", "updateStrategy", "rollingUpdate", "maxUnavailable"), nil, ""),
 			},
 		},
 		{
 			name: "greater than 100 percent maxUnavailable",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.ParallelPodManagement,
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					UpdateStrategy: apps.StatefulSetUpdateStrategy{
-						Type: apps.RollingUpdateStatefulSetStrategyType,
-						RollingUpdate: &apps.RollingUpdateStatefulSetStrategy{
-							MaxUnavailable: intStrAddr(intstr.FromString("101%")),
-						},
-					},
-				},
-			},
+			set: mkStatefulSet(&validPodTemplate,
+				tweakReplicas(3),
+				tweakUpdateStrategy(mkUpdateStrategy(
+					tweakMaxUnavailable(intstr.FromString("101%")),
+				)),
+			),
 			errs: field.ErrorList{
 				field.Invalid(field.NewPath("spec", "updateStrategy", "rollingUpdate", "maxUnavailable"), nil, ""),
 			},
 		},
 		{
 			name: "invalid ordinals.start ",
-			set: apps.StatefulSet{
-				ObjectMeta: metav1.ObjectMeta{Name: "abc-123", Namespace: metav1.NamespaceDefault},
-				Spec: apps.StatefulSetSpec{
-					PodManagementPolicy: apps.ParallelPodManagement,
-					UpdateStrategy:      apps.StatefulSetUpdateStrategy{Type: apps.RollingUpdateStatefulSetStrategyType},
-					Selector:            &metav1.LabelSelector{MatchLabels: validLabels},
-					Template:            validPodTemplate.Template,
-					Replicas:            3,
-					Ordinals:            &apps.StatefulSetOrdinals{Start: -2},
-				},
-			},
+			set: mkStatefulSet(&validPodTemplate,
+				tweakReplicas(3),
```

Comment:
This is an awesome example - is Replicas ACTUALLY important to this test case, or was it just set for no real reason?  It doesn't seem like it SHOULD matter.

This cleanup makes such things obvious, and now there can be followups to remove unnecessary noise  like this (if it is unnecessary :).  Followups, please - not the same PR.